Add app_id and app_metadata attributes (#90)
eProsima/DDS-Record-Replay: docs/rst/notes/forthcoming_version.rst

* Store *DDS Record & Replay* version in metadata record of the generated MCAP files.
* Move dynamic types storage from metadata to attachments MCAP section.
* Store schemas in OMG IDL format (instead of ROS 2 msg).

> * Store `app_id` and `app_metadata` attributes in  *DDS Record & Replay* participants.

Reviewer: ```suggestion
* Set `app_id` and `app_metadata` attributes on *DDS Record & Replay* participants.
```
